src/molecules/BottomSheet/BottomSheet.native.js

```diff
@@ -0,0 +1,190 @@
+  border-top-right-radius: 8px;
+  border-top-left-radius: 8px;
+  box-shadow: ${(props) => `0px -4px 15px ${props.theme.colors.primary[930]}`};
+  elevation: 4;
+`;
+
+const BottomSheetDragBar = styled(View)`
+  background-color: ${(props) => props.theme.colors.shade[920]};
+  border-radius: 4px;
+`;
+
+const BottomSheet = forwardRef(
+  (
+    {
+      snapPoint = DEFAULT_SNAP_POINT,
+      children,
+      onBackDropPress = () => {},
+      onBackButtonPress = () => {},
+      onOpened = () => {},
+      onClosed = () => {},
+      adjustToContentHeight = false,
+      alwaysOpen = 0,
+    },
+    ref,
+  ) => {
+    const theme = useTheme();
+    const [headerHeight, setHeaderHeight] = useState(0);
+    const [contentHeight, setContentHeight] = useState(0);
+    const bottomsheetChildrenGroupByType = reactChildrenGroupByType(children);
+
+    const headerComponent = bottomsheetChildrenGroupByType[Header];
+    const footerComponent = bottomsheetChildrenGroupByType[Footer];
+    const contentComponent = bottomsheetChildrenGroupByType[Content];
+
+    if (headerComponent?.length > 1) {
+      throw new Error(
+        `expected to have single \`BottomSheet.Header\` but found ${headerComponent.length}`,
+      );
+    }
+
+    if (footerComponent?.length > 1) {
+      throw new Error(
+        `expected to have single \`BottomSheet.Footer\` but found ${footerComponent.length}`,
+      );
+    }
+
+    const handleHeaderLayoutChange = useCallback((e) => {
+      setHeaderHeight(e.nativeEvent.layout.height);
+    }, []);
+
+    const handleContentLayoutChange = useCallback((e) => {
+      setContentHeight(e.nativeEvent.layout.height);
+    }, []);
+
+    let contentContainerHeight = DEFAULT_SNAP_POINT - headerHeight;
+    if (alwaysOpen > 0) {
+      contentContainerHeight = alwaysOpen - headerHeight;
+    }
+    const isScrollableContent = contentHeight > contentContainerHeight;
+
+    return (
+      <RNModalize
+        ref={ref}
+        snapPoint={snapPoint}
+        HeaderComponent={
+          <View onLayout={handleHeaderLayoutChange}>
+            <HeaderContainer>
+              <Flex alignItems="center">
+                <Space padding={[1, 0, 1.5, 0]}>
+                  <View>
+                    <Size height={0.5} width={8}>
+                      <BottomSheetDragBar />
+                    </Size>
+                  </View>
+                </Space>
+              </Flex>
+              {headerComponent?.length ? (
+                <>
+                  {headerComponent}
+                  <Divider color="shade.920" horizontal />
+                </>
+              ) : null}
+            </HeaderContainer>
+          </View>
+        }
+        FloatingComponent={
+          <Position position="absolute" left={0} right={0} bottom={0}>
+            <View>
+              {isScrollableContent && (
+                <LinearGradient
+                  locations={linearGradientLocations}
```

Comment:
what does location means

Reply:
These are the different % which liner-gradient accepts. W.r.t web context considering `linear-gradient(180deg, #F9FBFE 0%, rgba(249, 251, 254, 0.6) 29.69%, rgba(249, 251, 254, 0) 100%);`,  the percentage values what you see are the locations.